spotbugs/src/main/java/edu/umd/cs/findbugs/detect/UnreadFields.java

```diff
@@ -816,6 +817,32 @@ public void sawOpcode(int seen) {
         previousOpcode = seen;
     }
 
+    /**
+     *
+     * @return true if the method is considered to be an initializer method. Fields might be initialized outside of a constructor,
+     * for instance through JUnit's BeforeEach
+     */
+    private boolean isInitializerMethod() {
+        if ("init".equals(getMethodName()) || "initialize".equals(getMethodName())) {
+            return true;
+        }
+
+        for (AnnotationEntry a : getMethod().getAnnotationEntries()) {
+            String typeName = a.getAnnotationType();
+
+            if ("Ljakarta/annotation/PostConstruct;".equals(typeName)
```

Comment:
Could put these in an ordered array and use `Arrays.binarySearch`, or an `ArrayList` and use `contains`.